packages/atomic/src/components/facets/atomic-date-range/atomic-date-range.tsx

```diff
@@ -15,9 +15,24 @@ export class AtomicDateRange {
    * It can be expressed as a Javascript date, as a number using epoch time or as a string using the ISO 8601 format.
    */
   @Prop() public start!: Date | string | number;
+
+  @Watch('start')
```

Comment:
https://stenciljs.com/docs/reactive-data#watch-decorator

> The Watch decorator does not fire when a component initially loads.

The watch is meant for props that can change after the component is loaded, that's not the case here.
You would have to call the validation methods in the `initialize` method. You can refer to `atomic-sort-dropdown.tsx` for an example.